### PR TITLE
Display a editor gizmo icon for Listener3D

### PR DESCRIPTION
**Pull request by Calinou**
`master` version of https://github.com/godotengine/godot/pull/52758.

The icon was present in `editor/icons/`, but it was never implemented in the editor gizmos code. This gizmo's visibility can be toggled in the 3D viewport's **View** menu as usual.

This also removes some unused gizmo drawing code (overridden methods that are no longer called anywhere).

See https://github.com/godotengine/godot/pull/49266.

## Preview

*ReflectionProbe on the left, Listener3D on the right.*

image

### CODE REVIEW

**Review by YeldhamDev — APPROVED** [review]
Needs a rebase.

**Review by reduz — APPROVED** [review]

**Comment by reduz** [comment]
looks good, please rebase. I think it still needs approval from @JFonS

File: editor/node_3d_editor_gizmos.h [hunk 1]
```diff
@@ -70,6 +70,23 @@ class AudioStreamPlayer3DGizmoPlugin : public EditorNode3DGizmoPlugin {
...
+
+public:
+	bool has_gizmo(Node3D *p_spatial) override;
+	String get_gizmo_name() const override;
+	int get_priority() const override;
+
+	String get_handle_name(const EditorNode3DGizmo *p_gizmo, int p_idx) const override;
+	Variant get_handle_value(EditorNode3DGizmo *p_gizmo, int p_idx) const override;
+	void set_handle(EditorNode3DGizmo *p_gizmo, int p_idx, Camera3D *p_camera, const Point2 &p_point) override;
+	void commit_handle(EditorNode3DGizmo *p_gizmo, int p_idx, const Variant &p_restore, bool p_cancel = false) override;
```
Comment (JFonS) [inline editor/node_3d_editor_gizmos.h:84]: There's no need to override these methods if the gizmo doesn't have any handles.

**Review by JFonS — COMMENTED** [review]
Looks good, just needs a rebase and getting rid of some unused methods :)

**Comment by Calinou** [comment]
Rebased and tested again, it works successfully.

image

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**